x-pack/plugins/stack_connectors/server/connector_types/stack/opsgenie/connector.test.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import crypto from 'crypto';
+import { ActionsConfigurationUtilities } from '@kbn/actions-plugin/server/actions_config';
+import { actionsConfigMock } from '@kbn/actions-plugin/server/actions_config.mock';
+import { actionsMock } from '@kbn/actions-plugin/server/mocks';
+import { loggingSystemMock } from '@kbn/core-logging-server-mocks';
+import { MockedLogger } from '@kbn/logging-mocks';
+import { OpsgenieConnectorTypeId } from '.';
+import { OpsgenieConnector } from './connector';
+
+describe('OpsgenieConnector', () => {
+  let connector: OpsgenieConnector;
+  let mockedActionsConfig: jest.Mocked<ActionsConfigurationUtilities>;
+  let logger: MockedLogger;
+  let services: ReturnType<typeof actionsMock.createServices>;
+
+  const defaultCreateAlertExpect = {
+    method: 'post',
+    url: 'https://example.com/v2/alerts',
+    headers: { Authorization: 'GenieKey 123', 'Content-Type': 'application/json' },
+    responseSchema: expect.anything(),
+  };
+
+  const createCloseAlertExpect = (alias: string) => ({
+    method: 'post',
+    url: `https://example.com/v2/alerts/${alias}/close?identifierType=alias`,
+    headers: { Authorization: 'GenieKey 123', 'Content-Type': 'application/json' },
+    responseSchema: expect.anything(),
+  });
+
+  beforeEach(() => {
+    logger = loggingSystemMock.createLogger();
+    services = actionsMock.createServices();
+    mockedActionsConfig = actionsConfigMock.create();
+
+    connector = new OpsgenieConnector({
+      configurationUtilities: mockedActionsConfig,
+      config: { apiUrl: 'https://example.com' },
+      connector: { id: '1', type: OpsgenieConnectorTypeId },
+      secrets: { apiKey: '123' },
+      logger,
+      services,
+    });
+  });
+
+  it('calls request with the correct arguments for creating an alert', async () => {
+    const connectorSpy = jest
```

Comment:
I think is better to mock `axios` instead of the method of the class. This way if the sub-action framework introduces an unintended breaking change or a feature change your tests can catch it.